app/routes.py

```diff
@@ -1,2 +1,289 @@
+            "description": new_task.description,
+            "is_complete": new_task.completed_at is not None
+        }
+    }
+    return jsonify(response_body), 201
+
+
+@tasks_bp.route("", methods=["GET"])
+def read_task():
+
+    # tasks hold all the information from the database. Becasue Task is referring to a table. So we are capturing all the records(rows) in the table and holding it in tasks.
+    # queries are the records (i.e. rows)
+    # creating task_response variable
+    title_query = request.args.get("sort")
+    if title_query == "asc":
+        tasks = Task.query.order_by(Task.title.asc())
+    elif title_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+    else:
+        tasks = Task.query.all()
+
+    task_responses = []
+    for task in tasks:
+        task_response = {
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": task.completed_at is not None
+        }
+        if task.goal_id:
+            task_response["goal_id"] = task.goal_id
+        task_responses.append(task_response)
+    return jsonify(task_responses), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["GET"])
+def read_one_task(task_id):
+    task = Task.query.get_or_404(task_id)
+
+    task_response = {
+        "id": task.task_id,
+        "title": task.title,
+        "description": task.description,
+        "is_complete": task.completed_at is not None
+    }
+    if task.goal_id:
+        task_response["goal_id"] = task.goal_id
+
+    response_body = {"task": task_response}
+
+    return jsonify(response_body), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["PUT"])
+def update_one_task(task_id):
+    task = Task.query.get_or_404(task_id)
+
+    request_body = request.get_json()
+
+    task.title = request_body["title"]
+    task.description = request_body["description"]
+
+    db.session.commit()
+
+    response_body = {
+        "task": {
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": task.completed_at is not None
+        }
+    }
+    return jsonify(response_body), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["DELETE"])
+def delete_one_task(task_id):
+    task = Task.query.get_or_404(task_id)
+
+    db.session.delete(task)
+    db.session.commit()
+
+    return jsonify({
+        "details": f"Task {task.task_id} \"{task.title}\" successfully deleted"
+    }), 200
+
+
+def slack_bot(title):
+    query_path = {'channel': 'slack_api_test_channel', 'text': title}
+    header = {'Authorization': os.environ.get('BOT')}
+    response = requests.post(
+        'https://slack.com/api/chat.postMessage', params=query_path, headers=header)
+    return response.json()
+
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_complete(task_id):
+    task = Task.query.get_or_404(task_id)
+    task.completed_at = datetime.utcnow()
```

Comment:
I also like using `utcnow()` rather than `now()` for the datetime information. We didn't set up the datetime in the database to store timezone information, so when working with dates and time, it's usually a good idea to pass them around and store them in UTC, and only convert them to local time when needed for display (like in a UI). Working with dates and times is tricky!